src/transaction-signer.ts

```diff
@@ -0,0 +1,18 @@
+export class TransactionSigner {
```

Comment:
Why is this a class and not an interface?

Reply:
Using an interface is only typescript related, using class will be more compatible if someone wants to use the generated js code